ugs-core/src/com/willwinder/universalgcodesender/utils/Settings.java

```diff
@@ -24,16 +24,21 @@ This file is part of Universal Gcode Sender (UGS).
 import com.willwinder.universalgcodesender.pendantui.PendantConfigBean;
 import com.willwinder.universalgcodesender.types.Macro;
 import com.willwinder.universalgcodesender.types.WindowSettings;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 import java.util.*;
 import org.apache.commons.lang3.StringUtils;
 
 public class Settings {
     // Transient, don't serialize or deserialize.
     transient private SettingChangeListener listener = null;
+    transient public static int HISTORY_SIZE = 20;
 
     private String firmwareVersion = "GRBL";
     private String fileName = System.getProperty("user.home");
+    private ArrayDeque<String> fileHistory = new ArrayDeque<>();
```

Comment:
Really just a detail... =)

I usually want the member or a variable to have the most common interface specified which could make things simpler when for example refactoring:
```private Deque<String> fileHistory = new ArrayDeque<>();``